src/main/java/com/iota/iri/service/snapshot/impl/SnapshotServiceImpl.java

```diff
@@ -433,7 +433,7 @@ private void cleanupExpiredSolidEntryPoints(Tangle tangle, Map<Hash, Integer> ol
                 try {
                     // only clean up if the corresponding milestone transaction was cleaned up already -> otherwise
                     // let the MilestonePrunerJob do this
-                    if (TransactionViewModel.fromHash(tangle, transactionHash).getType() ==
+                    if (transactionPruner != null && TransactionViewModel.fromHash(tangle, transactionHash).getType() ==
```

Comment:
There is no point doing the null check here
You can do it right at the beginning of the method...

Reply:
changed it